hash.c

```diff
@@ -2044,22 +2055,22 @@ rb_hash_rehash_i(VALUE key, VALUE value, VALUE arg)
 
 /*
  *  call-seq:
- *     hsh.rehash -> hsh
- *
- *  Rebuilds the hash based on the current hash values for each key. If
- *  values of key objects have changed since they were inserted, this
- *  method will reindex <i>hsh</i>. If Hash#rehash is
- *  called while an iterator is traversing the hash, a
- *  RuntimeError will be raised in the iterator.
- *
- *     a = [ "a", "b" ]
- *     c = [ "c", "d" ]
- *     h = { a => 100, c => 300 }
- *     h[a]       #=> 100
- *     a[0] = "z"
- *     h[a]       #=> nil
- *     h.rehash   #=> {["z", "b"]=>100, ["c", "d"]=>300}
- *     h[a]       #=> 100
+ *     hsh.rehash -> self
+ *
+ *  Rebuilds the hash table based on the current hash value for each key;
```

Comment:
Some new users may find this terminology confusing. It's fault of Ruby using "Hash" to mean the map as well as the digest for indexing into the map.

Maybe saying "Rebuilds the hash table by recomputing the hash value for each key". I'd also be okay with calling it  "hash index for each key".

Reply:
Fixed.